x-pack/plugins/ml/public/application/components/data_grid/feature_importance/use_classification_path_data.tsx

```diff
@@ -165,9 +199,87 @@ export const buildClassificationDecisionPathData = ({
     }
     return undefined;
   });
+
+  // get the baseline for the current class from the trained_models metadata
+  const baselineClass = baselines.find(
+    (bl) => getStringBasedClassName(bl.class_name) === getStringBasedClassName(currentClass)
+  );
   const filteredFeatureImportance = mappedFeatureImportance.filter(
     (f) => f !== undefined
   ) as ExtendedFeatureImportance[];
 
-  return buildDecisionPathData(filteredFeatureImportance);
+  const finalResult: DecisionPathPlotData = filteredFeatureImportance
+    // sort by absolute importance so it goes from bottom (baseline) to top
+    .sort(
+      (a: ExtendedFeatureImportance, b: ExtendedFeatureImportance) =>
+        b.absImportance! - a.absImportance!
+    )
+    .map((d) => [d[FEATURE_NAME] as string, d[FEATURE_IMPORTANCE] as number, NaN]);
+
+  /**
+   * For binary classification
+   */
+  if (baselines.length === 2) {
+    // transform the numbers into the probability space
+    // starting with the baseline retrieved from trained_models metadata
+    let logOddSoFar = baselineClass?.baseline ? baselineClass?.baseline : 0;
+    for (let i = featureImportance.length - 1; i >= 0; i--) {
+      logOddSoFar += finalResult[i][1];
+      const predictionProbabilitySoFar = Math.exp(logOddSoFar) / (Math.exp(logOddSoFar) + 1);
+      finalResult[i][2] = predictionProbabilitySoFar;
+    }
+    return finalResult;
+  }
+
+  /**
+   * For multiclass classification
+   */
+
+  // first calculate the denominator
+  // (\sum_{x\in{A,B,C}} exp(baseline(x) + \sum_{i=0}^j feature_importance_i(x)))
+  let denominator = 0;
+  for (let x = 0; x < baselines.length; x++) {
+    // grab the feature importance for class x
+    const _featureImportanceOfClassX: ExtendedFeatureImportance[] = featureImportance
+      .map((feature) => {
+        const classFeatureImportance = Array.isArray(feature.classes)
+          ? feature.classes.find(
+              (c) =>
+                getStringBasedClassName(c.class_name) ===
+                getStringBasedClassName(baselines[x].class_name)
+            )
+          : feature;
+        if (
+          classFeatureImportance &&
+          typeof classFeatureImportance[FEATURE_IMPORTANCE] === 'number'
+        ) {
+          return {
+            [FEATURE_NAME]: feature[FEATURE_NAME],
+            [FEATURE_IMPORTANCE]: classFeatureImportance[FEATURE_IMPORTANCE],
+            absImportance: Math.abs(classFeatureImportance[FEATURE_IMPORTANCE] as number),
+          };
+        }
+        return undefined;
+      })
+      .filter((d) => d !== undefined) as ExtendedFeatureImportance[];
+
+    const featureImportanceOfClassX = _featureImportanceOfClassX.reduce(
+      (acc, a) => acc + a.importance!,
+      0
+    );
```

Comment:
As you are using `for` loop I assume the performance is a concern here. But right after you are using multiple array methods that cause extra iterations. 
I think you should just define `let featureImportanceOfClassX  = 0`, then iterate `featureImportance` with another nested `for` loop and increase it based on `absImportance` of each valid `classFeatureImportance`

Reply:
Updated here https://github.com/elastic/kibana/pull/80229/commits/10eae2d74c3867d9416aa3c29437da5d971cd485